### PR TITLE
Generate Grammar section of langref from zig-spec

### PR DESCRIPTION
**Pull request by thejoshwolfe**
The first commit adds anchors for the generator script to look for. The second commit brings the contents up to date with the latest zig-spec master branch.

### CODE REVIEW

**Comment by thejoshwolfe** [comment]
This game up from: https://github.com/ziglang/zig/pull/8063

See also https://github.com/ziglang/zig-spec/pull/24

File: doc/langref.html.in [hunk 1]
```diff
@@ -11040,20 +11040,26 @@ fn readU32Be() u32 {}
       {#header_close#}
 
       {#header_open|Grammar#}
-      <pre><code>Root &lt;- skip ContainerMembers eof
+      <p>
+      This content is sourced from the <a href="https://github.com/ziglang/zig-spec">zig-spec</a> project.
+      It is not necessarily up to date. Please make contributions if you find a problem.
+      </p>
+      <!-- $$BEGIN GENERATED GRAMMAR$$ --><pre><code>Root &lt;- skip ContainerMembers eof
```
Comment (andrewrk) [inline doc/langref.html.in:11047]: we already have a meta language inside here, can you use that instead of HTML comments?

**Comment by andrewrk** [comment]
Are you using a tool to do the update? Is it something to put in the tools/ directory?

File: doc/langref.html.in (hunk 1 above)
Comment (thejoshwolfe) [reply doc/langref.html.in:11047]: good question. that would mean embedding the contents of a file from another repo into this html document with some html escaping. is there a precedent for doing that? can you point me at the tool that generates the docs?

**Comment by andrewrk** [comment]
30+ days old draft; closing

**Pull request closed without merging**